src/Illuminate/Foundation/ComposerScripts.php

```diff
@@ -41,6 +41,17 @@ protected static function clearCompiled()
     {
         $laravel = new Application(getcwd());
 
+        // Create environment file if it doesn't exists.
+        if (! file_exists($laravel->environmentFilePath())) {
+            exec("php -r \"copy('".$laravel->environmentFilePath().".example',
+             '".$laravel->environmentFilePath().');"');
```

Comment:
> This operation is literally the same as when you create a new laravel application and new .env file is created after installation of all packages.

The operation is only the same when you're cloning `laravel/laravel`. What if you're cloning `yourorganization/project` repository which the team has decided to remove `.env.example`.

> the commands as listed here are only run when laravel installer is used to create the application.

The command only executed during `post-root-package-install`, `ComposerScripts::clearCompiled()` is execute during `post-install-cmd` and `post-update-cmd`.